woocommerce/woocommerce-ios
# Order Creation: Add customer summary screen

Part of: https://github.com/woocommerce/woocommerce-ios/issues/5412.

## Description

This PR adds customer data summary view and updates minor layout issues in customer section.
Note: Data on summary view have disclosure indicators, but it's not yet editable.

## Changes

- Adds `OrderCustomerSummary` view.
- Adds `NavigationRow` as reusable SwiftUI component.
- Updates `OrderCustomerSection` layout to have more consistent spacing.
- Updates `OrderCustomerSection` to display only shipping address + add bottom button.
- Updates `OrderCustomerSection` and `OrderStatusSection` to use safeAreaInsets from Environment.
- Updates `EditOrderAddressForm` to disable:
  - autocorrection on all fields
  - autocapitalization on email fied
  - (both were huge personal annoyance while testing, any considerations?)

## Testing 

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Tap "+ Add customer" button to display Address Form.
5. Enter some data.
6. Tap "Done" to save and close address form.
7. Confirm that customer section is updated with new data.
8. Tap "Edit" or "View Billing Information" in customer section.
9. Confirm that summary view shows correct data - billing and shipping addresses should be same.

## Screenshots
Empty state|Customer section|Customer summary
--|--|--
![Simulator Screen Shot - 1](https://user-images.githubusercontent.com/3132438/148279770-47b23954-0ac4-48ec-b213-66cb9af85c2c.png)|![Simulator Screen Shot - 2](https://user-images.githubusercontent.com/3132438/148279794-e510dfe0-3d0c-4432-93f3-b2d57f9edcf1.png)|![Simulator Screen Shot - 3](https://user-images.githubusercontent.com/3132438/148279805-bc50866f-6a60-4c61-b4aa-316e7e7f1e9e.png)

---
